### PR TITLE
refactor (jkube-kit/common-maven): Migrate all tests from JUnit4 to JUnit5 (#1551)

### LINKED ISSUE #1551: (jkube-kit-common-maven): Replace JUnit4 with JUnit5
### Component

jkube-kit-common-maven

### Task description

Child of #1546

- Remove JUnit4 dependencies
- Add JUnit5 dependencies
- Refactor tests

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description

Fixes #1551 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenConfigurationExtractorTest.java [hunk 1]
```diff
@@ -133,8 +131,9 @@ public void should_parse_list_of_mixed_elements() {
         final Map<String, Object> expected = new HashMap<>();
         expected.put("b", expectedC);
 
-        assertTrue(config.containsKey("a"));
-        assertEquals(expected, config.get("a"));
+        assertThat(config)
+                .containsKey("a")
```
Comment (rohanKanojia) [inline jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenConfigurationExtractorTest.java:135]: containsEntry should verify both scenarios. Let's remove this.
```suggestion
```

File: jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java [hunk 2]
```diff
@@ -13,73 +13,74 @@
...
-
+        assertThat(props).isNotNull().isEmpty();
     }
 
     @Test
-    public void testMultipleProfilesParsing() {
+    void testMultipleProfilesParsing() {
         Properties props = SpringBootUtil.getPropertiesFromApplicationYamlResource(null, getClass().getResource("/util/test-application-with-multiple-profiles.yml"));
-        assertTrue(props.size() > 0);
+        assertThat(props.size() > 0).isTrue();
```
Comment (rohanKanojia) [inline jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java:64]: You can remove this and directly use `.isNotEmpty()` in assertion below

**Comment by rohanKanojia** [comment]
Note: (To be done by maintainers) We're introducing a new dependency `org.mockito:mockito-junit-jupiter:4.6.1`. We would need to create CQ or use Eclipse Dash tool

File: jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenUtilTest.java [hunk 3]
```diff
@@ -42,32 +36,36 @@
...
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.mockito.Mockito.when;
-import static org.mockito.Mockito.verify;
+import static org.assertj.core.api.Assertions.entry;
 import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+@ExtendWith(MockitoExtension.class)
```
Comment (rohanKanojia) [inline jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenUtilTest.java:63]: Could you please not use `@ExtendWith` annotation and use manual instantiation instead?

Comment (anurag-rajawat) [reply jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenUtilTest.java:63]: Do you mean a setup method like this
```java
  @BeforeEach
  void setUp() {
    mavenProject = mock(MavenProject.class);
    pluginManager = mock(BuildPluginManager.class);
    log = mock(KitLogger.class);
  }
```
 or instantiating these fields in each test?

Comment (rohanKanojia) [reply jkube-kit/common-maven/src/test/java/org/eclipse/jkube/kit/common/util/MavenUtilTest.java:63]: You're right. I meant this. If these fields are specific to single test, maybe move this to individual test. If these fields are reused in all tests, we  can have them in set up method like you suggested.

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**